src/jit/morph.cpp

```diff
@@ -6051,14 +6057,15 @@ GenTreePtr Compiler::fgMorphField(GenTreePtr tree, MorphAddrContext* mac)
 {
     assert(tree->gtOper == GT_FIELD);
 
-    noway_assert(tree->gtFlags & GTF_GLOB_REF);
-
     CORINFO_FIELD_HANDLE symHnd          = tree->gtField.gtFldHnd;
     unsigned             fldOffset       = tree->gtField.gtFldOffset;
     GenTreePtr           objRef          = tree->gtField.gtFldObj;
     bool                 fieldMayOverlap = false;
     bool                 objIsLocal      = false;
 
+    noway_assert(((objRef != nullptr) && (objRef->IsLocalAddrExpr() != nullptr)) ||
+                 ((tree->gtFlags & GTF_GLOB_REF) != 0));
+
```

Comment:
This was previously a noway_assert above that was checking that it was marked `GTF_GLOB_REF`. I merely changed it to allow for the case where `objRef` is a local address expression. One might argue it should be a regular assert, but that's orthogonal to my change.